### DISCUSSION TITLE
CDNs to adopt JSR packages

### DISCUSSION THREAD

**Discussion by trymeouteh** [jsr-io/jsr#850] [Ideas]
Many CDNs get packages from NPM and make them available for web browsers to pull from.

Would be great if JSR was able to be adopted by CDNs.

**Comment by marvinhagemeister** [comment]
esm.sh is one of the CDNs that supports JSR packages, see https://github.com/esm-dev/esm.sh?tab=readme-ov-file#supported-registries

**Reply by trymeouteh** [reply]
Is it possible to use esm.sh to get scripts and stylesheets using the `<script>` and `<link>` tags instead of using `import`?

**Reply by frytg** [reply]
yes, should work using import maps: https://esm.sh/#docs -> How to Use

```js
<script type="importmap">
  {
    "imports": {
      "react": "https://esm.sh/react@18.2.0"
    }
  }
</script>
<script type="module">
  import React from "react" // alias to https://esm.sh/react@18.2.0
</script>
```

**Reply by trymeouteh** [reply]
Never heard of import maps before. Do import maps increase website loading speed compared to loading a javascript file since it has to import multiple module files instead of one single bundled script?

Is there a way to load a javascript file or CSS stylesheet file using ESM?

`<script src="https://esm.sh/path/to/jquery.min.js"></script>`

**Reply by trymeouteh** [reply]
Found a solution to use CSS and JS UMD scripts in HTML tags
https://github.com/esm-dev/esm.sh/discussions/993
